### DISCUSSION TITLE
My Octolamp build!

### DISCUSSION THREAD

**Discussion by bencurrandev** [martinwoodward/octolamp#26] [Show and tell]
Had a great time building this and putting it together, needed some tactical planning for the prints as the front and reflector were only a few hours, but the black outer wanted 22 hours at the 0.2 quality I usually print at, so I pushed it out to a 0.28 layer thickness to bring the print time down but not impact the quality. Came out great!
![IMG_9217](https://github.com/martinwoodward/octolamp/assets/75711425/12020328-6cff-4345-84da-e81bbc327656)
![IMG_9222](https://github.com/martinwoodward/octolamp/assets/75711425/c5fbed66-baea-4acc-a1ef-c800285610a3)
Learning how to get the most out of WLED has been a bunch of fun and now I know the resources are available easily and are affordable, I think I have a few more plans with them! 

https://github.com/martinwoodward/octolamp/assets/75711425/f959f6dc-6415-4c59-94e0-37b33c7e31fd

I think I might need to talk to you Martin about how to take the logo and build it into a 3d model and take it apart for 3D printing, because I should probably build an Azure logo version now 😆

**Comment by martinwoodward** [comment]
I include the .f3d file if you want to open it up in Fusion 360 and take a look.

That said, several things helped me about the Invertocat logo and gave me the idea to build a lamp this way as a simple click-fit model that you could light with a single uninterrupted strip. Not every logo will have charactertics that will make that quite as easy. The Invertocat logo also had the perfectly sized chunky bit in the middle to hide the electronics and somewhere to place the hanging holes so that it balanced easily. The only modification I had to make to the logo was to subtly flatten the bottom of the circle in the black outline so that this can sit on a shelf as well as hang on the wall.

However you should be able to figure something out, just might be a more complex build. If you need more than 100 LED's then your electronics get a tiny bit more complicated (you'll probably need a 3.3 to 5v level shifter for the data line and take your 5v supply direct from the inbound power rather than coming off the board)

For the snap fits and the reflector insert I did some experimentation with offsets so that there is decent tolerances for printing with different materials etc. Again, if you are doing a one-off you won't need to be as careful there but be prepared to do a few test prints along the way as that was probably the trickiest part to get right.

I'm def not a Fusion 360 expert, learned anything I do know from YouTube tutorials and I don't do enough modelling that I'll forget stuff between models and have to look up the same videos to remind me how to do some things.  But def have a play - just be prepared that it's a lot more work than just a simple SVG projection but it's good fun problem solving work too. I personally started by sketching out by hand where to store the electronics and what your led run will look like, then thought through the minimum components so that they hold themselves together and how to print without needing supports.

**Comment by martinwoodward** [comment]
Thinking about it, a technique like this might be much more appropriate and less work. Using bendy led strips was how I was originally going to go before I realised I could make a stand-alone printable thing due to the shape I had to play with: https://youtu.be/5t02s6qDqtE
